pylot/planning/frenet_optimal_trajectory/fot_planning_operator.py

```diff
@@ -0,0 +1,259 @@
+                    self._flags.carla_timeout), self._log_file_name)
+
+    def on_can_bus_update(self, msg):
+        self._logger.debug('@{}: received can bus message'.format(
+            msg.timestamp))
+        self._can_bus_msgs.append(msg)
+
+    def on_prediction_update(self, msg):
+        self._logger.debug('@{}: received prediction message'.format(
+            msg.timestamp))
+        self._prediction_msgs.append(msg)
+
+    def on_watermark(self, timestamp, waypoints_stream):
+        self._logger.debug('@{}: received watermark'.format(timestamp))
+
+        # get ego info
+        can_bus_msg = self._can_bus_msgs.popleft()
+        vehicle_transform = can_bus_msg.data.transform
+
+        # get obstacles
+        prediction_msg = self._prediction_msgs.popleft()
+        obstacle_list = self._build_obstacle_list(vehicle_transform,
+                                                  prediction_msg)
+        # update waypoints
+        if not self._waypoints:
+            self._update_global_waypoints(vehicle_transform)
+
+        # compute optimal frenet trajectory
+        path, csp, s0, initial_conditions = \
+            self._compute_optimal_frenet_trajectory(can_bus_msg, obstacle_list)
+
+        # send debug info
+        self._logger.debug("@{}: Initial conditions: {}"
+                           .format(timestamp, initial_conditions))
+        if path:
+            self._logger.debug("@{}: Frenet Path X: {}".format(timestamp,
+                                                               path.x))
+            self._logger.debug("@{}: Frenet Path Y: {}".format(timestamp,
+                                                               path.y))
+            self._logger.debug("@{}: Frenet Path V: {}".format(timestamp,
+                                                               path.s_d))
+
+        # construct and send waypoint message
+        waypoints_message = self._construct_waypoints(timestamp, path, csp, s0)
+        waypoints_stream.send(waypoints_message)
+
+    def _compute_optimal_frenet_trajectory(self, can_bus_msg, obstacle_list):
+        """
+        Compute the optimal frenet trajectory, given current environment info.
+        """
+        # convert waypoints to frenet coordinates
+        wx = []
+        wy = []
+        for wp in itertools.islice(self._waypoints, 0, DEFAULT_NUM_WAYPOINTS):
+            wx.append(wp.location.x)
+            wy.append(wp.location.y)
+        tx, ty, tyaw, tc, csp = generate_target_course(wx, wy)
+
+        # compute frenet optimal trajectory
+        s0, c_speed, c_d, c_d_d, c_d_dd = \
+            self._compute_initial_conditions(can_bus_msg, wx, wy, csp)
+        target_speed = (c_speed + self._flags.target_speed) / 2
+        path = frenet_optimal_planning(
+            csp, s0, c_speed, c_d, c_d_d, c_d_dd, obstacle_list, target_speed)
+
+        # log initial conditions for debugging
+        initial_conditions = {
```

Comment:
Why isn't the logging happening here, and instead you return the initial_conditions?

Reply:
Good point, fixed!